apps/taxes/src/modules/avatax/tax-code/avatax-tax-code-matches.service.ts

```diff
@@ -0,0 +1,40 @@
+import { AuthData } from "@saleor/app-sdk/APL";
+import { createUrqlClientFromCtx } from "../../../lib/graphql";
+import { Logger, createLogger } from "../../../lib/logger";
+import { createSettingsManager } from "../../app/metadata-manager";
+import {
+  AvataxTaxCodeMatch,
+  AvataxTaxCodeMatchRepository,
+  AvataxTaxCodeMatches,
+} from "./avatax-tax-code-match-repository";
+
+export class AvataxTaxCodeMatchesService {
+  private logger: Logger;
+  private taxCodeMatchRepository: AvataxTaxCodeMatchRepository;
+
+  constructor(authData: AuthData) {
+    this.logger = createLogger({ name: "AvataxTaxCodeMatchesService" });
+    const client = createUrqlClientFromCtx(authData);
```

Comment:
IMO client should be injected. We share client literally from the beginning to the end. 

ALSO its easier to test - to inject mocked client instead mocking `createUrqlClientFromCtx`

Reply:
What exactly do you mean by injecting? Extending a class that has a client?

Reply:
Also:

> inject mocked client

Why mocked?

Reply:
I mean that injecting improves testability

Your current solution, to test it, requires complicated test setup with `vi.mock`, where you mock factory etc.

With dependency injection you just have

```
class SomeService {
  private apiClient: Client

  constructor(client: Client) {
    this.apiClient = client;
  }
}

```

now with tests you can create any mocked client and pass it to the constructor. Tests will be simple and really "unit"

Reply:
Moreover if your service only uses e.g. 1 method, like `query`, your dependency is 

```
client: Pick<Client, 'query'>
```

then you need only to mock this

```
const mockedClient:Pick<Client, 'query'> =  {
query: vi.fn()
}

```

and pass it to the constructor (interface segregation principle)

Reply:
it's related to: https://github.com/saleor/apps/pull/564#discussion_r1232111102